frontend/views/containers/chatroom/file-attachment/ChatAttachmentPreview.vue

```diff
@@ -125,70 +112,85 @@ export default {
   },
   data () {
     return {
-      objectURLList: [],
-      settledURLList: [],
-      loadingBoxStyles: []
+      mediaObjectURLList: {
+        [CHATROOM_ATTACHMENT_TYPES.IMAGE]: [],
+        [CHATROOM_ATTACHMENT_TYPES.VIDEO]: []
+      },
+      settledMediaURLList: {
+        [CHATROOM_ATTACHMENT_TYPES.IMAGE]: [],
+        [CHATROOM_ATTACHMENT_TYPES.VIDEO]: []
+      },
+      config: {
+        CHATROOM_ATTACHMENT_TYPES: CHATROOM_ATTACHMENT_TYPES
+      }
     }
   },
   computed: {
-    shouldPreviewImages () {
-      return !this.attachmentList.some(attachment => {
-        return this.fileType(attachment) === 'non-image'
-      })
+    sortedAttachments () {
+      const collections = {
+        [CHATROOM_ATTACHMENT_TYPES.NON_MEDIA]: [],
+        [CHATROOM_ATTACHMENT_TYPES.IMAGE]: [],
+        [CHATROOM_ATTACHMENT_TYPES.VIDEO]: []
+      }
+
+      for (const entry of this.attachmentList) {
+        const fType = getFileType(entry.mimeType)
+        collections[fType].push(entry)
+      }
+
+      return collections
     },
-    allImageAttachments () {
-      return this.attachmentList.filter(entry => this.fileType(entry) === 'image')
+    hasMediaAttachments () {
+      return this.sortedAttachments[CHATROOM_ATTACHMENT_TYPES.IMAGE].length > 0 ||
+        this.sortedAttachments[CHATROOM_ATTACHMENT_TYPES.VIDEO].length > 0
     },
     isPending () {
       return this.variant === MESSAGE_VARIANTS.PENDING
     },
     isFailed () {
       return this.variant === MESSAGE_VARIANTS.FAILED
+    },
+    canDelete () {
+      return this.isMsgSender || this.isGroupCreator
     }
   },
   mounted () {
-    if (this.shouldPreviewImages) {
-      const promiseToRetrieveURLs = this.attachmentList.map(attachment => this.getAttachmentObjectURL(attachment))
-      Promise.all(promiseToRetrieveURLs).then(urls => {
-        this.objectURLList = urls
-      })
-
-      if (this.isForDownload) {
-        this.loadingBoxStyles = this.attachmentList.map(attachment => {
-          return this.getStretchedDimension(attachment.dimension)
-        })
+    if (this.hasMediaAttachments) {
+      for (const mediaType of [CHATROOM_ATTACHMENT_TYPES.IMAGE, CHATROOM_ATTACHMENT_TYPES.VIDEO]) {
+        const attachments = this.sortedAttachments[mediaType]
+        if (attachments.length > 0) {
+          const promiseToRetrieveURLs = attachments.map(attachment => this.getAttachmentObjectURL(attachment))
+          Promise.all(promiseToRetrieveURLs).then(urls => {
+            this.mediaObjectURLList[mediaType] = urls
+          })
+        }
       }
 
       sbp('okTurtles.events/on', DELETE_ATTACHMENT, this.deleteAttachment)
     }
   },
   beforeDestroy () {
-    if (this.shouldPreviewImages) {
+    if (this.hasMediaAttachments) {
```

Comment:
It'd be a good idea here to add logic to abort active downloads to conserve memory and data.

However, this needs changes in Chelonia (I can add that if you prefer that). If this isn't implemented right now, can we at least add a `TODO` comment here?